.github/workflows/pre-release-nethermind.yml

```diff
@@ -0,0 +1,153 @@
+        dotnet-version: 7
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v2
+    - name: Set up Docker Buildx
+      id: buildx
+      uses: docker/setup-buildx-action@v2
+    - name: Build Nethermind.Runner
+      id: build-runner
+      run: |
+        build_timestamp=$(date '+%s')
+        cd nethermind && commit_hash=$(git describe --always --exclude=* --abbrev=40) && cd ..
+        echo "BUILD_TIMESTAMP=$build_timestamp" >> $GITHUB_OUTPUT
+        echo "COMMIT_HASH=$commit_hash" >> $GITHUB_OUTPUT
+        echo "COMMIT_HASH_SHORT=$(echo $commit_hash | awk '{print substr($0,0,8);}')" >> $GITHUB_OUTPUT
+        $SCRIPTS_PATH/build-runner.sh ${{ github.event.inputs.tag }} $commit_hash $build_timestamp
+    - name: Build Nethermind.Cli
+      run: $SCRIPTS_PATH/build-cli.sh ${{ github.event.inputs.tag }} ${{ steps.build-runner.outputs.COMMIT_HASH }} ${{ steps.build-runner.outputs.BUILD_TIMESTAMP }}
+    - name: Build Nethermind.Launcher
+      run: $SCRIPTS_PATH/build-launcher.sh
+    - name: Build Nethermind.Launcher for Linux arm64
+      run: |
+        cd nethermind
+        docker buildx build --platform=linux/arm64 -t tmp-launcher -f Dockerfile.launcher . --load
+        docker run --platform=linux/arm64 -v $PWD:/opt/mount --rm tmp-launcher bash -c "cp /nethermind/Nethermind.Launcher /opt/mount/"
+        mv Nethermind.Launcher $GITHUB_WORKSPACE/$PUB_DIR/linux-arm64/Nethermind.Launcher
+    - name: Archive packages
+      id: archive
+      env:
+        PACKAGE_PREFIX: nethermind-${{ github.event.inputs.tag }}-${{ steps.build-runner.outputs.COMMIT_HASH_SHORT }}
+      run: |
+        echo "PACKAGE_PREFIX=$PACKAGE_PREFIX" >> $GITHUB_OUTPUT
+        $SCRIPTS_PATH/archive-packages.sh
+    - name: Cache packages
+      uses: actions/cache@v3
+      with:
+        path: |
+          ${{ github.workspace }}/${{ env.PACKAGE_DIR }}/
+          ${{ env.SCRIPTS_PATH }}/
+        key: packages-${{ steps.build-runner.outputs.COMMIT_HASH }}
+
+  trigger-publish:
+    name: Trigger publish event to different sources
+    runs-on: ubuntu-latest
+    needs: build
+    environment:
+      name: Releases
+      url: https://github.com/NethermindEth/nethermind/releases/tag/${{ github.event.inputs.tag }}
+    steps:
+    - run: echo "Just a middle-man job"
+
+  publish-github:
+    name: Publish packages to GitHub Releases
+    runs-on: ubuntu-latest
+    needs: trigger-publish
+    steps:
+    - name: Restore packages from cache
+      uses: actions/cache@v3
+      with:
+        path: |
+          ${{ github.workspace }}/${{ env.PACKAGE_DIR }}/
+          ${{ env.SCRIPTS_PATH }}/
+        key: packages-${{ needs.build.outputs.COMMIT_HASH }}
+    - name: Publish
+      env:
+        GIT_COMMIT: ${{ needs.build.outputs.COMMIT_HASH }}
+        GIT_TAG: ${{ github.event.inputs.tag }}
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        PACKAGE_PREFIX: ${{ needs.build.outputs.PACKAGE_PREFIX }}
+      run: $SCRIPTS_PATH/publish-github.sh
```

Comment:
You should probably update the github script and make it a pre-release or RC release. Not sure what's possible.